app-foundation/4-projects/development/boa_gke.tf

```diff
@@ -66,14 +78,22 @@ module "boa_gke_project" {
 }
 
 # Service account to allow  Bank of Anthos Pods to securely communicate with GCP APIs, in specific Cloud SQL and Cloud Operations
-module "boa_gsa_sa" {
-  source     = "terraform-google-modules/service-accounts/google"
-  version    = "~> 3.0"
-  project_id = module.boa_gke_project.project_id
-  names      = ["boa-gsa"]
-  project_roles = [
-    "${module.boa_gke_project.project_id}=>roles/cloudtrace.agent",
-    "${module.boa_gke_project.project_id}=>roles/monitoring.metricWriter",
-    "${module.boa_sql_project.project_id}=>roles/cloudsql.client"
-  ]
+resource "google_service_account" "boa_gsa_sa" {
+  account_id  = "boa-gsa"
+  description = "Service account to allow Bank of Anthos Pods to securely communicate with GCP APIs, in specific Cloud SQL and Cloud Operations"
+  project     = module.boa_gke_project.project_id
+}
+
+resource "google_project_iam_member" "boa_gsa_sa_roles_gke" {
+  for_each = toset(local.gsa_sa_roles.gke)
+  project  = module.boa_gke_project.project_id
+  role     = each.value
+  member   = "serviceAccount:${google_service_account.boa_gsa_sa.email}"
+}
+
+resource "google_project_iam_member" "boa_gsa_sa_roles_sql" {
+  for_each = toset(local.gsa_sa_roles.sql)
+  project  = module.boa_sql_project.project_id
+  role     = each.value
+  member   = "serviceAccount:${google_service_account.boa_gsa_sa.email}"
```

Comment:
can we for_each over `project_id=[roles]`

Reply:
resolved offline - keeping as is due to `The "for_each" value depends on resource attributes that cannot be determined
until apply`